### DISCUSSION TITLE
Repeat vs Rearrange: Is there anything rearrange does that repeat cannot?

### DISCUSSION THREAD

**Discussion by Interpause** [arogozhnikov/einops#243] [Q&A] [answered]
Besides the whole readability thing, I noticed that `repeat` seems able to perform all ops `rearrange` does. Can I ask if there is anything I am missing?

**Comment by arogozhnikov** [comment, marked as answer]
Hi @Interpause 
That's correct. `reduce` can also work as `rearrange`. 
It is purely readability thing: when you see `rearrange`, you know number of elements stays the same.

**Reply by Interpause** [reply]
I see, I also noticed through testing `rearrange` throws errors whenever you try and mess with the number of dimensions. Thanks.
